### DISCUSSION TITLE
raise AttributeError("The API key is required for new authorizations. "

### DISCUSSION THREAD

**Discussion by fallensamael** [kalanakt/All-Url-Uploader#111] [Q&A]
I need help with correctly specifying variables. I followed the instructions to modify Uploader/config.py and try for app.json, but I encountered an error in both cases. Can you please assist me in identifying the issue?

**Comment by jemalas** [comment]
how to solved?

**Reply by kalanakt** [reply]
Se𝘯𝘵 me logs pls

**Comment by kalanakt** [comment]
can you provide me full of log? the error may be the wrong Telegram API or Pyrogram version, try to update the program version.

```shell
pip install pyrogram --upgrade
```

**Reply by Amansain** [reply]
I face same problem pls help me

**Reply by kalanakt** [reply]
can you send me full logs file

**Reply by Amansain** [reply]
Welcome to Ubuntu 22.04.2 LTS (GNU/Linux 5.19.0-1025-aws x86_64)
 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage
  System information as of Wed Jul 26 02:20:23 UTC 2023
  System load:  0.013671875       Processes:             105
  Usage of /:   64.9% of 7.57GB   Users logged in:       1
  Memory usage: 35%               IPv4 address for eth0: 172.31.17.97
  Swap usage:   0%
 * Ubuntu Pro delivers the most comprehensive open source security and
   compliance features.
   https://ubuntu.com/aws/pro
Expanded Security Maintenance for Applications is not enabled.
0 updates can be applied immediately.
9 additional security updates can be applied with ESM Apps.
Learn more about enabling ESM Apps service at https://ubuntu.com/esm
*** System restart required ***
Last login: Wed Jul 26 02:17:39 2023 from 49.42.95.94
ubuntu@ip-172-31-17-97:~$ cd All-Url-Uploader
ubuntu@ip-172-31-17-97:~/All-Url-Uploader$ pip3 install -U -r requirements.txt
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: aiohttp in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 1)) (3.8.5)
Requirement already satisfied: beautifulsoup4 in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 2)) (4.12.2)
Requirement already satisfied: hachoir in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 3)) (3.2.0)
Requirement already satisfied: numpy in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 4)) (1.25.1)
Requirement already satisfied: Pillow in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 5)) (10.0.0)
Requirement already satisfied: pyrogram==2.0.43 in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 6)) (2.0.43)
Requirement already satisfied: requests in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 7)) (2.31.0)
Requirement already satisfied: tgcrypto in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 8)) (1.2.5)
Requirement already satisfied: olefile in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 9)) (0.46)
Requirement already satisfied: motor in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 10)) (3.2.0)
Requirement already satisfied: pymongo in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 11)) (4.4.1)
Requirement already satisfied: dnspython in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 12)) (2.4.0)
Requirement already satisfied: psutil in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 13)) (5.9.5)
Requirement already satisfied: loggers in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 14)) (0.1.4)
Requirement already satisfied: filetype in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 15)) (1.2.0)
Requirement already satisfied: tldextract in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 16)) (3.4.4)
Requirement already satisfied: aiofiles in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 17)) (23.1.0)
Requirement already satisfied: pyromod in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 18)) (2.0.0)
Requirement already satisfied: yt-dlp in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 19)) (2023.7.6)
Requirement already satisfied: filesplit==4.0.0 in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 20)) (4.0.0)
Requirement already satisfied: mega.py in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 21)) (1.0.8)
Requirement already satisfied: opencc in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 22)) (1.1.6)
Requirement already satisfied: wget in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 23)) (3.2)
Requirement already satisfied: youtube_dl in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 24)) (2021.12.17)
Requirement already satisfied: Flask==2.2.2 in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 25)) (2.2.2)
Requirement already satisfied: gunicorn==20.1.0 in /home/ubuntu/.local/lib/python3.10/site-packages (from -r requirements.txt (line 26)) (20.1.0)
Requirement already satisfied: pyaes==1.6.1 in /home/ubuntu/.local/lib/python3.10/site-packages (from pyrogram==2.0.43->-r requirements.txt (line 6)) (1.6.1)
Requirement already satisfied: pysocks==1.7.1 in /home/ubuntu/.local/lib/python3.10/site-packages (from pyrogram==2.0.43->-r requirements.txt (line 6)) (1.7.1)
Requirement already satisfied: Werkzeug>=2.2.2 in /home/ubuntu/.local/lib/python3.10/site-packages (from Flask==2.2.2->-r requirements.txt (line 25)) (2.3.6)
Requirement already satisfied: Jinja2>=3.0 in /usr/lib/python3/dist-packages (from Flask==2.2.2->-r requirements.txt (line 25)) (3.0.3)
Requirement already satisfied: itsdangerous>=2.0 in /home/ubuntu/.local/lib/python3.10/site-packages (from Flask==2.2.2->-r requirements.txt (line 25)) (2.1.2)
Requirement already satisfied: click>=8.0 in /usr/lib/python3/dist-packages (from Flask==2.2.2->-r requirements.txt (line 25)) (8.0.3)
Requirement already satisfied: setuptools>=3.0 in /usr/lib/python3/dist-packages (from
gunicorn==20.1.0->-r requirements.txt (line 26)) (59.6.0)
Requirement already satisfied: aiosignal>=1.1.2 in /home/ubuntu/.local/lib/python3.10/site-packages (from aiohttp->-r requirements.txt (line 1)) (1.3.1)
Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in /home/ubuntu/.local/lib/python3.10/site-packages (from aiohttp->-r requirements.txt (line 1)) (4.0.2)
Requirement already satisfied: attrs>=17.3.0 in /usr/lib/python3/dist-packages (from aiohttp->-r requirements.txt (line 1)) (21.2.0)
Requirement already satisfied: multidict<7.0,>=4.5 in /home/ubuntu/.local/lib/python3.10/site-packages (from aiohttp->-r requirements.txt (line 1)) (6.0.4)
Requirement already satisfied: yarl<2.0,>=1.0 in /home/ubuntu/.local/lib/python3.10/site-packages (from aiohttp->-r requirements.txt (line 1)) (1.9.2)
Requirement already satisfied: frozenlist>=1.1.1 in /home/ubuntu/.local/lib/python3.10/site-packages (from aiohttp->-r requirements.txt (line 1)) (1.4.0)
Requirement already satisfied: charset-normalizer<4.0,>=2.0 in /home/ubuntu/.local/lib/python3.10/site-packages (from aiohttp->-r requirements.txt (line 1)) (3.2.0)
Requirement already satisfied: soupsieve>1.2 in /home/ubuntu/.local/lib/python3.10/site-packages (from beautifulsoup4->-r requirements.txt (line 2)) (2.4.1)
Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/lib/python3/dist-packages (from requests->-r requirements.txt (line 7)) (1.26.5)
Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests->-r requirements.txt (line 7)) (3.3)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3/dist-packages (from requests->-r requirements.txt (line 7)) (2020.6.20)
Requirement already satisfied: httpcore>=0.17.3 in /home/ubuntu/.local/lib/python3.10/site-packages (from dnspython->-r requirements.txt (line 12)) (0.17.3)
Requirement already satisfied: sniffio<2.0,>=1.1 in /home/ubuntu/.local/lib/python3.10/site-packages (from dnspython->-r requirements.txt (line 12)) (1.3.0)
Requirement already satisfied: requests-file>=1.4 in /home/ubuntu/.local/lib/python3.10/site-packages (from tldextract->-r requirements.txt (line 16)) (1.5.1)
Requirement already satisfied: filelock>=3.0.8 in /usr/lib/python3/dist-packages (from
tldextract->-r requirements.txt (line 16)) (3.6.0)
Requirement already satisfied: pycryptodomex in /home/ubuntu/.local/lib/python3.10/site-packages (from yt-dlp->-r requirements.txt (line 19)) (3.18.0)
Requirement already satisfied: mutagen in /home/ubuntu/.local/lib/python3.10/site-packages (from yt-dlp->-r requirements.txt (line 19)) (1.46.0)
Requirement already satisfied: brotli in /home/ubuntu/.local/lib/python3.10/site-packages (from yt-dlp->-r requirements.txt (line 19)) (1.0.9)
Requirement already satisfied: websockets in /home/ubuntu/.local/lib/python3.10/site-packages (from yt-dlp->-r requirements.txt (line 19)) (11.0.3)
Requirement already satisfied: pathlib==1.0.1 in /home/ubuntu/.local/lib/python3.10/site-packages (from mega.py->-r requirements.txt (line 21)) (1.0.1)
Requirement already satisfied: pycryptodome<4.0.0,>=3.9.6 in /home/ubuntu/.local/lib/python3.10/site-packages (from mega.py->-r requirements.txt (line 21)) (3.18.0)
Requirement already satisfied: tenacity<6.0.0,>=5.1.5 in /home/ubuntu/.local/lib/python3.10/site-packages (from mega.py->-r requirements.txt (line 21)) (5.1.5)
Requirement already satisfied: anyio<5.0,>=3.0 in /home/ubuntu/.local/lib/python3.10/site-packages (from httpcore>=0.17.3->dnspython->-r requirements.txt (line 12)) (3.7.1)
Requirement already satisfied: h11<0.15,>=0.13 in /home/ubuntu/.local/lib/python3.10/site-packages (from httpcore>=0.17.3->dnspython->-r requirements.txt (line 12)) (0.14.0)
Requirement already satisfied: six in /usr/lib/python3/dist-packages (from requests-file>=1.4->tldextract->-r requirements.txt (line 16)) (1.16.0)
Requirement already satisfied: MarkupSafe>=2.1.1 in /home/ubuntu/.local/lib/python3.10/site-packages (from Werkzeug>=2.2.2->Flask==2.2.2->-r requirements.txt (line 25)) (2.1.3)
Requirement already satisfied: exceptiongroup in /home/ubuntu/.local/lib/python3.10/site-packages (from anyio<5.0,>=3.0->httpcore>=0.17.3->dnspython->-r requirements.txt (line 12)) (1.1.2)
ubuntu@ip-172-31-17-97:~/All-Url-Uploader$ python3 bot.py
Traceback (most recent call last):
  File "/home/ubuntu/All-Url-Uploader/bot.py", line 53, in <module>
    app.run()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pyrogram/methods/utilities/run.py", line 84, in run
    self.start()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pyrogram/sync.py", line 66, in async_to_sync_wrap
    return loop.run_until_complete(coroutine)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pyrogram/methods/utilities/start.py", line 58, in start
    is_authorized = await self.connect()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pyrogram/methods/auth/connect.py", line 40, in connect
    await self.load_session()
  File "/home/ubuntu/.local/lib/python3.10/site-packages/pyrogram/client.py", line 574, in load_session
    raise AttributeError("The API key is required for new authorizations. "
AttributeError: The API key is required for new authorizations. More info: https://docs.pyrogram.org/start/auth
ubuntu@ip-172-31-17-97:~/All-Url-Uploader$

**Reply by prisonerr627** [reply]
got the same issue

**Reply by prisonerr627** [reply]
i have tried to export as environment variables and also hardcode the values in the config.py but still the same issue

**Comment by kalanakt** [comment]
- [x] fix and tested
